server/api/social/index.js

```diff
@@ -27,8 +27,10 @@ router.get('/social/list/:id', async (req, res, next) => {
     const col = await dbRef.doc(username).collection('social').get();
     const replyObj = {};
     col.docs.forEach((d) => {
-      const { isLinked } = d.data();
-      if (isLinked) replyObj[d.id] = Validate.filterSocialPlatformPublic({ ...d.data() });
+      const { isLinked, isPublic } = d.data();
+      if (isLinked && isPublic !== false) {
```

Comment:
why not `isPrivate` ? what is the default value for `isPublic`?
if it is default public, explicitly setting `isPrivate = true` seems better?